File: src/src_platform_jfx/fmg/jfx/utils/ImgUtils.java
package fmg.jfx.utils;

import javafx.scene.SnapshotParameters;
import javafx.scene.canvas.Canvas;
import javafx.scene.image.Image;
import javafx.scene.image.ImageView;
import javafx.scene.paint.Color;

/** вспомогательный класс для преобразований картинок */
public final class ImgUtils {

   public static Image toImage(Canvas self) {
      SnapshotParameters params = new SnapshotParameters();
      params.setFill(Color.TRANSPARENT);
      return self.snapshot(params, null);
   }

   public static Image zoom(Image img, int newWidth, int newHeight) {
      ImageView imageView = new ImageView(img);
      imageView.setPreserveRatio(true);
      imageView.setFitWidth(newWidth);
      imageView.setFitHeight(newHeight);
      SnapshotParameters params = new SnapshotParameters();
      params.setFill(Color.TRANSPARENT);
      return imageView.snapshot(params, null);
  }

}

File: src/src_platform_android/fmg/android/utils/ImgUtils.kt
package fmg.android.utils

import android.graphics.Bitmap

/** вспомогательный класс для преобразований картинок  */
//object ImgUtils {

    /** change size Bitmap  */
    fun Bitmap.zoom(newWidth: Int, newHeight: Int): Bitmap? {
        return if (this == null) null else Bitmap.createScaledBitmap(this, newWidth, newHeight, false)
    }

//}
